node-cart-exercise-SABETTA_definitivo

SkyronVarn opened the pull request:
non sono sicuro di aver mandato il codice quando ero a scuola, nel dubbio lo rimando per evitare casini
Grazie per la visione e buona giornata/serata

SkyronVarn committed:
cart-receipt-exercise-Sabetta

non funziona totalmente, non calcola il totale scontato perché non trova la percentuale sconto dallo User
non penso di riuscire a fare di meglio con la mia attuale conoscenza di JS

SkyronVarn committed:
Update core-cart.mjs

SkyronVarn committed:
node-cart-receipt_SABETTA-Definitivo

Questa versione funziona, non come quella che consegnai precedentemente

danielebarbaro commented on package.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "dependencies": {
+    "chalk": "^5.0.1"
```
Cosa ci fa questo file qui?

SkyronVarn replied:
L'ho notato ora, ieri ho installato Chalk da npm per il progetto del Wordle
ho capito stamattina perché non mi trovava il Node Module, era finito tra questi file.
Nella consegna corretta poi tolgo i file futili

danielebarbaro commented on package-lock.json:
```diff
@@ -0,0 +1,30 @@
+{
```
Cosa ci fa questo file qui?

danielebarbaro commented on core-cart.mjs:
```diff
@@ -3,8 +3,80 @@ import {carts, products, promoCode, users} from "./dataset.mjs";
+return somma;
+}
+
+const createDelimiter = (openClose, symbol, times) => '+ -------------------------------------------------- +';
```
dovrebbero essere autogenerati almeno i trattini.
Prova ad usare https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/repeat

danielebarbaro commented:
@SkyronVarn  

- per favore controlla bene i calcoli
- non stampare ricevute se il carrello non ha prodotti o lo user non ha abbastanza soldi

Se vuoi ambire al massimo dei voti e vuoi generare le ricevute in formato txt, crea una cartella chiamata receipts nel progetto e leggi .

Io nella mia soluzione ho usato:
```js
const printReceipt = (content, filename) => fs.writeFileSync(filename, content);
```

dove content è il contenuto di una singola ricevuta.

SkyronVarn committed:
Cart-Receipt-Exercise-Sabetta

non sono riuscito a stampare le ricevute in file, tuttavia il resto del programma sembra funzionare per il meglio

danielebarbaro commented on core-cart.mjs:
```diff
@@ -3,8 +3,87 @@ import {carts, products, promoCode, users} from "./dataset.mjs";
+const createDelimiter3 = (openClose, symbol, times) => '** ' + '-'.repeat(50) + ' **';
+
+const calculateWallet = (wallet, totale) => wallet - totale;
+const assicuredWallet = (tot) => tot >= 0 ? "'s remaining credit: " + tot.toFixed(2) + "€": false ;
```
? `remaining credit:` è scritto nel readme?
